code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -70,6 +70,13 @@
 			update_inv_ears()
 		if(slot_glasses)
 			glasses = I
+			var/obj/item/clothing/glasses/G = I
+			if(G.tint)
+				update_tinttotal()
+			if(G.vision_correction)
+				update_vision_overlays()
+			if(G.vision_flags || G.darkness_view || G.invis_override || G.invis_view)
+				update_sight()

```

Comment:
we update our vision when equippind and unequipping glasses that changes our vision.